Develop2

xotparis opened the pull request:
Создание API-тестов для кейсов:
Создание пользователя;
Логин пользователя;
Изменение данных пользователя;
Создание заказа;
Получение заказов конкретног пользователя;
С подключением allure-отчета.

Commit:
Создание API-тестов для кейсов:
Создание пользователя;
Логин пользователя;
Изменение данных пользователя;
Создание заказа;
Получение заказов конкретног пользователя;

Commit:
Add Allure test results

Commit:
Add Allure test results

Commit:
Создание API-тестов для кейсов:
Создание пользователя;
Логин пользователя;
Изменение данных пользователя;
Создание заказа;
Получение заказов конкретног пользователя;

Commit:
Небольшие правки

pomegranate282922 commented on src/main/java/models/User.java:
```diff
@@ -0,0 +1,32 @@
+
+    public void setName(String name) {
+        this.name = name;
+    }
```
Можно применить библиотеку Lombok для уменьшения boilerplate кода 
Можно улучшить

pomegranate282922 commented on src/test/java/edu/praktikum/Diplom_2/helpers/CreateUser.java:
```diff
@@ -0,0 +1,41 @@
+                .post(CREATE_ENDPOINT);
+    }
+
+}
```
Отлично что есть рандомизация

pomegranate282922 commented on src/test/java/edu/praktikum/Diplom_2/helpers/LoginUser.java:
```diff
@@ -0,0 +1,17 @@
+package edu.praktikum.Diplom_2.helpers;
```
Пакеты именуем строчными буквами без _ 
Нужно исправить

pomegranate282922 commented on src/test/java/edu/praktikum/Diplom_2/tests/BaseTest.java:
```diff
@@ -0,0 +1,40 @@
+
+        if (accessToken != null) {
+
+            given()
```
Все вызовы ручек нужно вынести из тестовых классов. 
нужно исправить

pomegranate282922 commented on src/test/java/edu/praktikum/Diplom_2/tests/BaseTest.java:
```diff
@@ -0,0 +1,40 @@
+
+        }
+    }
+}
```
Отлично что есть иерархия тестовых классов

pomegranate282922 commented on src/test/java/edu/praktikum/Diplom_2/tests/CreateOrderTests.java:
```diff
@@ -0,0 +1,94 @@
+public class CreateOrderTests extends BaseTest {
+    private User validUser;
+
+    Order order = new Order(List.of("61c0c5a71d1f82001bdaaa6d"));
```
Валидные хэши не хардкодим, а получаем динамически из соответствующей ручки. 
нужно исправить

pomegranate282922 reviewed (commented):
Работа хорошая! Все тесты реализованы. Есть рандомизация. Везде удаляется юзер. 
замечания:
1. Все вызовы ручек вынести из тестовых классов 
2. Нейминг 
3. Валидные ингредиенты берем из ручки

Commit:
Подключил библиотеку lombok.
Вынес ручку по удалениею пользователя из теста.
Сделал ручку на получение ингридиентов, убрав хардкод из тестов.

pomegranate282922 commented on src/test/java/edu/praktikum/diplom2/tests/ChangeUserDataTests.java:
```diff
@@ -0,0 +1,80 @@
+    @Description("Positive case: Code 200")
+    public void changeUserData() {
+
+        Response updateResponse = given()
```
Все вызовы ручек нужно вынести в отдельные классы 
Нужно исправить

pomegranate282922 reviewed (commented):
Все отлично, но не все ручки вынесены - в итоге код выглядит неоднотипно, что нехорошо. Доделай, пожалуйста.

Commit:
Вынес ручки ChangeUserDataTests в отдельные классы

pomegranate282922 reviewed (approved):
Ок, зачет!
